rules/proto_rules.build_defs

```diff
@@ -395,10 +395,15 @@ def _protoc_rule(name, srcs, deps, language, plugin, protoc_flags, root_dir, pre
         cmd += ' ${SRCS}'
 
     cmd = f'(export OUT_DIR=$TMP_DIR/{language_out_dir} && {cmd})'
+    cmd += f' && (mv -f {language_out_dir}/${{PKG_DIR}}/* {out_dir}; true)'
 
-    cmd += f' && (mv -f {language_out_dir}/${{PKG_DIR}}/* {out_dir}; true) && (mv -f {language_out_dir}/* {out_dir}; true)'
+    if language_out_dir == 'go':
+        # Uses find+xargs+mv to flatten the paths and move the *.pb.go files.
+        cmd += f' && (find {language_out_dir} -name "*.pb.go" | xargs -i mv -f {{} {out_dir}; true)'
```

Comment:
We can probably just set `out_dir` to the package directory? 

If protoc outputs `api.pb.go` and `api_grpc.pb.go` to `go/test.com/src/proto/api`, setting  `output_dirs = ["go/test.com/src/proto/api/**"]` would work right?

Reply:
Tried it, your code doesn't work. The reason is that my config.Go.ImportPath=**example.com**, and the proto has
```
option go_package = "example.com/foo/bar/api";
```
Now protoc will generate files like plz-out/tmp/foo/bar/api/_grpc_go#protoc_go._build/go/**example.com**/foo/bar/api/api_grpc.pb.go, while {pkg_name} only matches "foo/bar/api" without prefix "example.com", so the files will not be copied.